samwise.js

```diff
@@ -10,19 +10,18 @@
     "use strict";
```

Comment:
SCRIPT-I'm a bit concerned about the potential performance impact of this loop. Have you thought about optimizing it?